### PR TITLE
progress bar component added (#154)

### PR DESCRIPTION
**Pull request by mrtmrtmlck**
Associated Issue: #154 

### Summary of Changes

- Progress Bar component has been added
- Styling of progress bar
- Tests

### CODE REVIEW

File: src/js/components/progress-bar.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { createComponent } from '../mini';
+import { connect } from '../store';
+import { totalRebusesCount } from '../rebuses';
+
+export function ProgressBar() {
+  return connect(
+    createComponent({
+      render() {
```
Comment (ollelauribostrom) [inline src/js/components/progress-bar.js]: You can use the props that will be passed to the render method instead,
```js
render({ rebuses }) {....
```

and then figure out how many rebuses that has been answered by doing,
```js
const answeredRebuses = rebuses.filter(rebus => rebus.isAnswered);
```

this way we don't have to get anything from localStorage, but can instead rely on the appStores state

File: src/js/rebuses.js [hunk 2]
```diff
@@ -631,3 +631,5 @@ export function getRebuses() {
     };
   });
 }
+
+export const totalRebusesCount = rebuses.length;
```
Comment (ollelauribostrom) [inline src/js/rebuses.js]: If we use the props passed to the render method instead in the ProgressBar component, we don't need to export/import this 🙂

File: src/css/main.css [hunk 3]
```diff
@@ -149,6 +149,37 @@ body,
   transform: translate3d(-50%, -50%, 0);
 }
 
+.progress-bar {
```
Comment (ollelauribostrom) [inline src/css/main.css]: These styles are a great start, to make the work in Firefox (and to be more in line with the rest of the app colorwise) refactor them to look something like:

```css
.progress-bar {
  grid-row-start: 4;
  grid-column-start: 1;
  grid-column-end: 6;
  border: none;
  background: #9c0629;
}

progress[value] {
  width: auto;
  height: 20px;
  -webkit-appearance: none;
}

progress[value]::-webkit-progress-bar {
  background-color: #9c0629;
}

progress[value]::-moz-progress-bar {
  background-color: #438365;
}

progress[value]::-webkit-progress-value {
  background-color: #438365;
}
```

Comment (mrtmrtmlck) [reply src/css/main.css]: Thanks for your comments :) I will take care as soon as possible.

**Comment by mrtmrtmlck** [comment]
@ollelauribostrom It's done! Again, thanks for your feedback :blush:

**Comment by ollelauribostrom** [comment]
Nice, thank you! 👍

**Pull request merged**